### PR TITLE
Add utilities for CHiME-7

### PR DESCRIPTION
**Pull request by boeddeker**

### CODE REVIEW

File: meeteval/io/stm.py [hunk 1]
```diff
@@ -66,55 +72,31 @@ def serialize(self):
...
-                p.pretty(list(self.lines))
+    def _load(cls, file_descriptor) -> 'List[STMLine]':
+        return [
+            STMLine.parse(line)
+            for line in file_descriptor
+            if len(line.strip()) > 0 and not line.strip().startswith(';')
+        ]
 
     @classmethod
     def merge(cls, *stms) -> 'STM':
```
Comment (thequilo) [inline meeteval/io/stm.py:90]: Does it make sense to move `merge` into the base class?

File: meeteval/io/base.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+@dataclass(frozen=True)
+class BaseLine:
+    @classmethod
+    def parse(cls, line: str) -> 'Self':
+        raise NotImplementedError(cls)
+
+    def serialize(self):
+        raise NotImplementedError(type(self))
+
+    def replace(self, **kwargs):
```
[the remark this answers is not in the document]
Comment (boeddeker) [reply meeteval/io/base.py:24]: done

File: meeteval/io/uem.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+      file
+
+    Since all field names from dscore don't match the typically used fienames
+    from NIST and the source code of
+    https://github.com/usnistgov/SCTK/blob/master/src/asclite/core/uemfilter.h
+    uses the typical names, use fild names that match with STM.
+
+    """
+    filename: str
+    channel: 'str | int'= '<NA>'
```
[the remark this answers is not in the document]
Comment (boeddeker) [reply meeteval/io/uem.py]: done

File: meeteval/io/uem.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    - file id  --  file id
+    - channel  --  channel (1-indexed)
+    - onset  --  onset of evaluation region in seconds from beginning of file
+    - offset  --  offset of evaluation region in seconds from beginning of
+      file
+
+    Since all field names from dscore don't match the typically used fienames
+    from NIST and the source code of
+    https://github.com/usnistgov/SCTK/blob/master/src/asclite/core/uemfilter.h
+    uses the typical names, use fild names that match with STM.
```
[the remark this answers is not in the document]
Comment (boeddeker) [reply meeteval/io/uem.py:32]: done

File: meeteval/io/base.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        else:
+            raise NotImplementedError(type(item), item)
+
+    def __iter__(self):
+        return iter(self.lines)
+
+    def __len__(self):
+        return len(self.lines)
+
+    def groupby(self, key) -> Dict[str, 'Self']:
```
[the remark this answers is not in the document]
Comment (boeddeker) [reply meeteval/io/base.py:93]: Done, I allow str, tuple and list

File: meeteval/io/chime7.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    ])
+
+
+def to_stm(*jsons, uem=None, file=sys.stdout):
+    if uem is not None:
+        uem = UEM.load(uem)
+
+    for json in jsons:
+        json = Path(json)
+        stm = json_to_stm(_load_json(json), json.with_suffix('').name)
```
[the remark this answers is not in the document]
Comment (boeddeker) [reply meeteval/io/chime7.py:85]: fixed

File: meeteval/io/stm.py (hunk 1 above)
Comment (boeddeker) [reply meeteval/io/stm.py:90]: I just kept merge, because it was already implemented. Is it somewhere used?

**Review by thequilo — APPROVED** [review]

File: meeteval/io/stm.py (hunk 1 above)
Comment (thequilo) [reply meeteval/io/stm.py:90]: I think not anymore. It was used in one of the `load` functions

**Pull request merged**